core.d.ts

```diff
@@ -15,6 +15,21 @@ interface ResourceObj<T> {
 	[index: string]: T
 }
 
+interface MimemessageFactory {
+	factory(cs: BatchFactory): CompiledBatch
+}
+
+interface BatchFactory {
+	body: string | Array<BatchFactory>
+	// This should be just string. But apparently there is currently no way to specify such a signature in TS
+	[index: string]: string | Array<BatchFactory>
```

Comment:
What is this indexing? I'm guessing headers? If so can you change this to `[headerName: string]: ...` (or a similar descriptive name)